### PR TITLE
Chapter 14: Following users

### PR DESCRIPTION
**Pull request by hvd1024**

### CODE REVIEW

**Comment by hvd1024** [comment]
screenshot from 2018-09-25 01-31-49

**Comment by hvd1024** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -42,6 +44,20 @@ def destroy
     redirect_to users_url
   end
 
+  def following
+    @title = "Following"
```
Comment (hhhhuy112) [inline app/controllers/users_controller.rb]: I18n em

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -42,6 +44,20 @@ def destroy
...
+    @user = User.find(params[:id])
+    @users = @user.following.paginate(page: params[:page])
+    render "show_follow"
+  end
+
+  def followers
+    @title = "Followers"
```
Comment (hhhhuy112) [inline app/controllers/users_controller.rb]: I18n em

File: app/models/relationship.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (hhhhuy112) [inline app/models/relationship.rb]: User.name thay cho "User"

File: app/models/user.rb [hunk 4]
```diff
@@ -27,19 +37,71 @@ def digest string
     def new_token
       SecureRandom.urlsafe_base64
     end
+
+    def select_activated
+      User.where(activated: true)
```
Comment (hhhhuy112) [inline app/models/user.rb]: viết scope em

File: app/models/user.rb [hunk 5]
```diff
@@ -27,19 +37,71 @@ def digest string
...
+  end
+
+  def password_reset_expired?
+    reset_sent_at < Settings.model.time_rs.hours.ago
+  end
+
+  def feed
+    following_ids = "SELECT followed_id FROM relationships
+                      WHERE  follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
```
Comment (hhhhuy112) [inline app/models/user.rb]: viết scope em

**Comment by hhhhuy112** [comment]
commented

File: app/models/user.rb (hunk 5 above)
Comment (hvd1024) [reply app/models/user.rb]: done

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (hvd1024) [reply app/controllers/users_controller.rb]: done

File: app/models/relationship.rb (hunk 3 above)
Comment (hvd1024) [reply app/models/relationship.rb]: done

File: app/models/user.rb (hunk 4 above)
Comment (hvd1024) [reply app/models/user.rb]: done

**Comment by hvd1024** [comment]
ready

**Comment by hhhhuy112** [comment]
merged

**Pull request merged**